Countries Info/ViewControllers/CountryDetailsViewController.swift

```diff
@@ -4,15 +4,144 @@
 //
 //  Created by Oleh Stasiv on 08.12.2020.
 //
-
 import UIKit
 
 class CountryDetailsViewController: UIViewController {
-
+    
+    
+    @IBOutlet weak var nameLabel: UILabel!
+    @IBOutlet weak var capitalLabel: UILabel!
+    @IBOutlet weak var regionLabel: UILabel!
+    @IBOutlet weak var populationLabel: UILabel!
+    
+    @IBOutlet weak var currenciesStack: UIStackView!
+    @IBOutlet weak var languagesStack: UIStackView!
+    @IBOutlet weak var timesZonesStack: UIStackView!
+    @IBOutlet weak var callingCodesStack: UIStackView!
+    
+    @IBOutlet var colorViews: [UIView]!
+    
     var viewModel: CountryDetailsViewModelProtocol!
     
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
+        nameLabel.text = viewModel.countryName
+        capitalLabel.text = viewModel.capital
+        regionLabel.text = viewModel.region
+        populationLabel.text = viewModel.population
+        setValue(array: viewModel.currencies, stack: currenciesStack, color: #colorLiteral(red: 1, green: 0.8941176471, blue: 0.6431372549, alpha: 1), type: CurrenciesButton.self)
+        setValue(array: viewModel.languages, stack: languagesStack, color: #colorLiteral(red: 0.9843137255, green: 0.8274509804, blue: 0.7529411765, alpha: 1), type: LanguagesButton.self)
+        setValue(array: viewModel.timeZones, stack: timesZonesStack, color: #colorLiteral(red: 0.7254901961, green: 0.9176470588, blue: 0.7647058824, alpha: 1), type: TimeZonesButton.self)
+        setValue(array: viewModel.callingCodes, stack: callingCodesStack, color: #colorLiteral(red: 0.6509803922, green: 0.9019607843, blue: 0.9921568627, alpha: 1), type: CallingCodesButton.self)
+        
+        setUI()
+    }
+    
+    private func setUI() {
+        addBackButton()
+        colorViews.forEach { $0.layer.cornerRadius = $0.frame.height / 2 }
+    }
+    
+    private func setValue(array: [String], stack: UIStackView, color: UIColor, type: UIButton.Type) {
+        let button = type.init()
+        setupButton(button)
+        
+        var width: CGFloat = button.intrinsicContentSize.width + Constants.spacing
+        configureStack(stack)
+        
+        for text in array {
+            let label = PaddingLabel(withInsets: Constants.padding, Constants.padding, Constants.padding, Constants.padding)
+            configureLabel(label: label, text: text, color: color)
+            width += label.intrinsicContentSize.width + Constants.spacing
+            
+            if width > stack.frame.width {
+                stack.addArrangedSubview(button)
+                addSpasingView(to: stack)
+                return
```

Comment:
Please think if there is another way of writing this logic, having `return` inside `for` loop is risky and may be error prone in the future, if someone will change this method implementation, thanks